apps/src/templates/instructions/TopInstructions.jsx

```diff
@@ -619,222 +559,115 @@ class TopInstructions extends Component {
-                      if (ref) {
-                        this.instructions = ref;
-                      }
-                    }}
+                    ref={ref => this.setInstructionsRef(ref)}
                     handleClickCollapser={this.handleClickCollapser}
                     adjustMaxNeededHeight={this.adjustMaxNeededHeight}
-                    isEmbedView={this.props.isEmbedView}
-                    teacherViewingStudentWork={
-                      this.state.teacherViewingStudentWork
-                    }
+                    isEmbedView={isEmbedView}
+                    teacherViewingStudentWork={teacherViewingStudentWork}
                   />
                 )}
-              {!this.props.hasContainedLevels &&
+              {!hasContainedLevels &&
                 isCSDorCSP &&
-                this.state.tabSelected === TabType.INSTRUCTIONS && (
-                  <div>
-                    <Instructions
-                      ref={ref => {
-                        if (ref) {
-                          this.instructions = ref;
-                        }
-                      }}
-                      longInstructions={this.props.longInstructions}
-                      onResize={this.adjustMaxNeededHeight}
-                      inTopPane
-                    />
-                  </div>
+                tabSelected === TabType.INSTRUCTIONS && (
+                  <Instructions
+                    ref={ref => this.setInstructionsRef(ref)}
+                    longInstructions={longInstructions}
+                    onResize={this.adjustMaxNeededHeight}
+                    inTopPane
+                  />
                 )}
             </div>
-            {this.state.tabSelected === TabType.RESOURCES && (
+            {tabSelected === TabType.RESOURCES && (
               <HelpTabContents
                 ref={ref => (this.helpTab = ref)}
-                videoData={videoData}
-                mapReference={this.props.mapReference}
-                referenceLinks={this.props.referenceLinks}
+                videoData={levelVideos ? levelVideos[0] : []}
+                mapReference={mapReference}
+                referenceLinks={referenceLinks}
               />
             )}
-            {displayFeedback && !this.state.fetchingData && (
+            {displayFeedback && !fetchingData && (
               <TeacherFeedback
-                user={this.props.user}
-                visible={this.state.tabSelected === TabType.COMMENTS}
+                user={user}
+                visible={tabSelected === TabType.COMMENTS}
                 displayKeyConcept={displayKeyConcept}
                 disabledMode={
-                  this.props.viewAs === ViewType.Student ||
-                  !this.state.teacherViewingStudentWork
+                  this.isViewingAsStudent() || !teacherViewingStudentWork
                 }
-                rubric={this.state.rubric}
+                rubric={rubric}
```

Comment:
there are a few more refs to declare in this chunk of code.

Reply:
I don't understand what you mean?

Reply:
can you declare `helpTab`, `commentTab`,  and `teacherOnlyTab` like you did with `instructions`?

Reply:
Oh - got it, you mean initialize them as null.